src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -323,8 +322,7 @@ private static boolean isSingleLineStatement(DetailAST statement) {
      */
     private static boolean isSingleLineWhile(DetailAST literalWhile) {
         boolean result = false;
-        if (literalWhile.getParent().getType() == TokenTypes.SLIST
-                && literalWhile.getLastChild().getType() != TokenTypes.SLIST) {
```

Comment:
We don't care if it has an `SLIST` or not as we only want ones without SLIST`s. This is showing the visitToken has a no `SLIST` check that isn't being checked first and is causing a lot of unnecessary checks in the rest of the code. I will make another PR for this after this is merged.
This happens quite a bit in this check.